Mod1/Lessons/Week5/Scope.md

```diff
@@ -0,0 +1,232 @@
+
+With this new definition in mind, we can see why the above code examples caused errors.
+
+Example 1:
+![Scope Example 1 With Highlighting](/Mod1/Images/Week5/ScopeExample1Highlighting.png)
+
+We are trying to access the `username` variable outside of the scope where it is defined.
+
+Example 2:
+![Scope Example 2 With Highlighting](/Mod1/Images/Week5/ScopeExample2Highlighting.png)
+
+Note that a variable is accessible **everywhere** inside the code block in which it's defined, even from within a nested code block.
+
+## Practice
+
+Here is a modified version of some code from our Class Interaction Lesson. 
+
+>Take 3 minutes: For each variable defined, determine the box for the scope where that variable is available.
+
+<details><summary>Solution</summary><br/>
+
+![Scope Example 3 With Highlighting](/Mod1/Images/Week5/ScopeExample3.png)
+</details>
+
+```c#
+Person stevie = new Person("Stevie");
+
+class Person
+{
+    public string Name { get; }
+    public List<Pet> Pets { get; }
+
+    public Person(string name)
+    {
+        Name = name;
+        Pets = new List<Pet>();
+    }
+
+    public void AddPet(Pet pet)
+    {
+        Pets.Add(pet);
+    }
+
+    public List<string> ListPetsNames()
+    {
+        var petNames = new List<string>();
+
+        foreach (var pet in Pets)
+        {
+            petNames.Add(pet.Name);
+            if (pet.Name == "Fluffy")
+            {
+                Console.WriteLine("Fluffy is my favorite name!")
+                Console.WriteLine($"We can access {petNames.Count} here too!")
+
+            }
+        }
+
+        return petNames;
+    }
+}
+```
+
+## Fixing Scope Errors
+
+With your partner, take 15 minutes to walk through this training that introduces a common pattern for fixing scope errors, then discuss the questions below.
+
+https://learn.microsoft.com/en-us/training/modules/csharp-code-blocks/2-exercise-variable-scope
+
+> In your own words, why did the code in step 2 have an error? </br> In your own words, how did you fix the scope error created in step 2? </br> What would happen if scope didn't exist? What problems might this cause?
+
+
+<!-- OPEN QUESTION: Any reason for students to know the difference between method, class, block scope. Seems to all act the same. Curly brackets cause a new code block with a new scope. -->
+
+
+
+
+<!-- Access modifiers is quite a nuanced topic and I'm inclided to separate it from scope. Maybe into a 30 minute mini lesson during projct work times? -->
+
+# Access Modifiers
+
+We are already familiar with including access modifiers when adding properties to a class. 
+```c#
+public string UserName;
+private string Role;
+```
+But when would you use `public` vs `private`? Why are access modifiers so helpful? Let's see an example!
+
+## Why are access modifiers helpful?
+
+Imagine that you're working on the development team for a bank. Your teammate Isabella wrote the following BankAccount class. 
+
+```c#
+class BankAccount
```

Comment:
I really like this bank example - its a good use case that is a real-world scenario 👍